File: web-app/src/modules/bookstore/services/bookstore-service.ts
import { Observable } from "rxjs";
import { Book } from "@app/modules/bookstore/models/book";

export interface BookstoreService {
	getBooksFromPeterStore(): Observable<Book[]>;
	getBooksFromGretaStore(): Observable<Book[]>;
	getBooksFromAllStore(): Observable<Book[]>;
}

File: api/Modules/Bookstore/Services/BookstoreService.cs
using API.Modules.Bookstore.Models;
using API.Modules.Bookstore.Repositories;

namespace API.Modules.Bookstore.Services
{
    public class BookstoreService : IBookstoreService
    {
        private readonly IBookstoreRepository bookstoreRepository;

        public BookstoreService(IBookstoreRepository bookstoreRepository)
        {
            this.bookstoreRepository = bookstoreRepository;
        }
        public IEnumerable<Book> GetAllPeterStoreBooks()
        {
            return bookstoreRepository.GetAllPeterStoreBooks();
        }

        public IEnumerable<Book> GetAllGretaStoreBooks()
        {
            return bookstoreRepository.GetAllGretaStoreBooks();
        }
    }
}
